lib/crono_trigger/worker.rb

```diff
@@ -145,5 +157,37 @@ def handle_signal_from_rdb
     rescue => ex
       CronoTrigger::GlobalExceptionHandler.handle_global_exception(ex)
     end
+
+    def monitor
+      return unless ActiveSupport::Notifications.notifier.listening?(CronoTrigger::Events::MONITOR)
+
+      CronoTrigger::Models::Worker.connection_pool.with_connection do
+        if CronoTrigger.workers.where("polling_model_names = ?", @model_names.to_json).order(:worker_id).limit(1).pluck(:worker_id).first != @crono_trigger_worker_id
+          # Return immediately to avoid redundant instruments
+          return
+        end
+
+        @model_names.each do |model_name|
+          model = model_name.classify.constantize
+          executable_count = model.executables.count
+
+          execute_lock_column = model.crono_trigger_column_name(:execute_lock)
+          oldest_execute_lock = model.executables(including_locked: true).where.not(execute_lock_column => 0).order(execute_lock_column).limit(1).pluck(execute_lock_column).first
+
+          next_execute_at_column = model.crono_trigger_column_name(:next_execute_at)
+          oldest_next_execute_at = model.executables.order(next_execute_at_column).limit(1).pluck(next_execute_at_column).first
+
+          now = Time.now
+          ActiveSupport::Notifications.instrument(CronoTrigger::Events::MONITOR, {
+            model_name: model_name,
+            executable_count: executable_count,
+            max_lock_duration_sec: oldest_execute_lock.nil? ? 0 : now.to_i - oldest_execute_lock,
+            max_latency_sec: oldest_next_execute_at.nil? ? 0 : now - oldest_next_execute_at,
```

Comment:
I think if all records have future time on `next_execute_at` column (that situation is a normal state), `max_latency_sec` returns a negative value.
Is that intentional?

Reply:
If the next_execute_at of all records is a future time, I think the result is nil, and the latency will be 0. I'll add a test later for the case.

Reply:
Ah, `executable` scope rejects future `next_execute_at`. I missed that.
There is no problem. Thanks!

Reply:
I added the test https://github.com/joker1007/crono_trigger/pull/39/commits/21a24be8b8e951f7f5697115086ec7aaf8ada5db.